sw/device/lib/dif/dif_uart.c

```diff
@@ -472,3 +472,58 @@ dif_uart_result_t dif_uart_tx_bytes_available(const dif_uart_t *uart,
 
   return kDifUartOk;
 }
+
+dif_uart_result_t dif_uart_fifo_reset(const dif_uart_t *uart,
+                                      dif_uart_fifo_reset_t reset) {
+  if (uart == NULL) {
+    return kDifUartBadArg;
+  }
+
+  uint32_t reg = mmio_region_read32(uart->base_addr, UART_FIFO_CTRL_REG_OFFSET);
+  bitfield_field32_t field = {
+      .mask = 0x1, .value = 1,
+  };
+
+  if (reset == kDifUartFifoResetRx || reset == kDifUartFifoResetAll) {
+    field.index = UART_FIFO_CTRL_RXRST;
+    reg = bitfield_set_field32(reg, field);
+  }
+
+  if (reset == kDifUartFifoResetTx || reset == kDifUartFifoResetAll) {
+    field.index = UART_FIFO_CTRL_TXRST;
+    reg = bitfield_set_field32(reg, field);
+  }
+
+  mmio_region_write32(uart->base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
+
+  return kDifUartOk;
+}
+
+dif_uart_result_t dif_uart_loopback_set(const dif_uart_t *uart,
+                                        dif_uart_loopback_t loopback,
+                                        dif_uart_enable_t enable) {
+  if (uart == NULL) {
+    return kDifUartBadArg;
+  }
+
+  bitfield_field32_t field = {
+      .mask = 0x1,
+  };
+  if (loopback == kDifUartLoopbackSystem) {
+    field.index = UART_CTRL_SLPBK;
+  } else {
+    field.index = UART_CTRL_LLPBK;
+  }
```

Comment:
just write `.field = loopback ? foo : bar;`

Reply:
I think I prefer using `?` with clearly `1`/`0`, `true`/`false` or `enable`/`disable` parameter, but I see your point.

Reply:
Not a big fan of `?` for non `0`/`1`, `true`/`false` or `enable/disable`, but I admit it looks cleaner, so I fixed it up.

Reply:
Same. I think people overuse it. My general rule is "if it fits on one line, use `?`". If the condition were long I might be tempted to write
```c
bool is_loopback = .....;
is_loopback ? FOO : BAR;
```

But, again, it's really in the eye of the beholder.